public/app/features/auth-config/fields.tsx

```diff
@@ -620,6 +627,69 @@ export function fieldMap(provider: string): Record<string, FieldData> {
         'If enabled, Grafana will match the Hosted Domain retrieved from the Google ID Token against the Allowed Domains list specified by the user.',
       type: 'checkbox',
     },
+    serverDiscoveryUrl: {
+      label: 'OpenID Connect Discovery URL',
+      description:
+        'The .well-known/openid-configuration endpoint for your IdP. The info extracted from this URL will be used to populate the Auth URL, Token URL and API URL fields.',
+      type: 'custom',
+      content: (setValue, watch) => {
+        const modalIsOpen = watch('serverDiscoveryModal');
```

Comment:
I think having `serverDiscoveryModal` on the state via the `useState` hook will make this easier as this field is not meant to be sent to the server. That way you also won't need the `watch` param here. 

Also I think this content could be a React component taking `setValue` as a prop.

Reply:
I removed the `ServerDiscoveryModal` field and used a state variable via `useState` instead. I also moved all content into its own React component `ServerDiscoveryField`.